components/eamxx/src/share/field/field_alloc_prop.hpp

```diff
@@ -61,21 +61,37 @@ namespace scream
 
 // Helper struct to store info related to the subviewing process
 struct SubviewInfo {
-  SubviewInfo () = default;
-  SubviewInfo (const int dim, const int slice, const int extent, const bool is_dynamic) :
-    dim_idx(dim), slice_idx(slice), dim_extent(extent), dynamic(is_dynamic) {}
-  SubviewInfo (const SubviewInfo&) = default;
-  SubviewInfo& operator= (const SubviewInfo&) = default;
-
-  int dim_idx    = -1;  // Dimension along which slicing happened
-  int slice_idx  = -1;  // Slice along dimension $dim_idx
-  int dim_extent = -1;  // Extent of dimension $dim_idx
-  bool dynamic   = false; // Whether this is a dynamic subview (slice_idx can change)
+  SubviewInfo() = default;
+  SubviewInfo(const int dim, const int slice, const int extent,
+              const bool is_dynamic)
+      : dim_idx(dim), slice_idx(slice), dim_extent(extent),
+        dynamic(is_dynamic) {}
+  // multi-slice subview across contiguous indices
+  SubviewInfo(const int dim, const int slice_beg, const int slice_end,
```

Comment:
As I said in another comment, let's remove the `dynamic` option from this overload. I can't see when/why it would be needed.